Bigtable/src/BigtableClient.php

```diff
@@ -0,0 +1,193 @@
+            [
+                'typeUrl' => 'type.googleapis.com/google.bigtable.admin.instance.v2.UpdateInstanceMetadata',
+                'callable' => function ($instance) {
+                    $name = InstanceAdminClient::parseName($instance['name'])['instance'];
+                    return $this->instance($name, $instance);
+                }
+            ], [
+                'typeUrl' => 'type.googleapis.com/google.bigtable.admin.instance.v2.CreateInstanceMetadata',
+                'callable' => function ($instance) {
+                    $name = InstanceAdminClient::parseName($instance['name'])['instance'];
+                    return $this->instance($name, $instance);
+                }
+            ]
+        ]);
+    }
+
+    /**
+     * Lazily instantiate an instance.
+     *
+     * Example:
+     * ```
+     * $instance = $bigtable->instance('my-instance');
+     * ```
+     *
+     * @param string $instanceId The instance ID
+     * @param array $instance [optional] {
+     *     Configuration options
+     *
+     *     @type string $displayName **Defaults to** the value of $instanceId.
+     *     @type int $instanceType Possible values are represented by the following constants:
+     *           `Google\Cloud\Bigtable\Instance::INSTANCE_TYPE_PRODUCTION` and
+     *           `Google\Cloud\Bigtable\Instance::INSTANCE_TYPE_DEVELOPMENT`.
+     *           **Defaults to** using `Google\Cloud\Bigtable\Instance::INSTANCE_TYPE_UNSPECIFIED`.
+     *     @type array $labels as key/value pair ['foo' => 'bar']. For more information, see
+     *           [Using labels to organize Google Cloud Platform resources](https://cloudplatform.googleblog.com/2015/10/using-labels-to-organize-Google-Cloud-Platform-resources.html).
+     * }
+     * @return Instance
+     */
+    public function instance($instanceId, array $instance = [])
+    {
+        return new Instance(
+            $this->connection,
+            $this->lroConnection,
+            $this->lroCallables,
+            $this->projectId,
+            $instanceId,
+            $instance
+        );
+    }
+
+    /**
+     * Constructs an array that contains cluster metadata.
+     *
+     * Example:
+     * ```
+     * $cluster = $bigtable->clusterMetadata('my-cluster', 'us-east-b');
+     * ```
+     *
+     * @param string $locationId The location ID
+     * @param int $storageType The storage media type for persisting Bigtable data. Possible values include
+     *            `Google\Cloud\Bigtable\Instance::STORAGE_TYPE_SSD`,
+     *            `Google\Cloud\Bigtable\Instance::STORAGE_TYPE_HDD` and
+     *            `Google\Cloud\Bigtable\Instance::STORAGE_TYPE_UNSPECIFIED`.
+     *            **Defaults to** `Google\Cloud\Bigtable\Instance::STORAGE_TYPE_UNSPECIFIED`.
+     * @param int $serveNodes
+     *
+     * @return array
+     */
+    public function clusterMetadata($clusterId, $locationId, $storageType = null, $serveNodes = null)
```

Comment:
This is more like a utility function, which could be made static, as it doesn't use any of the object properties.

Reply:
I disagree. Without a strong case in favor of making something static, it should not be. What would the use case be for making this statically accessible? It isn't a useful method outside the context of an existing instance of BigTableClient.

Reply:
It looks as though it is also intended to be used [here](https://github.com/GoogleCloudPlatform/google-cloud-php/pull/1158/files#diff-4c098f0913ba618d9a6b0428882eb248R207). I prefer static as well.

Reply:
My link may be outdated with all the changes being pushed. I'm having a hard time verifying on mobile. In any case, it was referincing `Instance::create()`, which a user could have access to an instance of without the `BigtableClient`. Additionally I agree with the point that since it doesn't touch any of the instance's scope it makes more sense static. As for naming I would prefer something which describes the action it performs, e.g. `buildClusterData` or the like. That way it won't be confused with the methods which eventually give us access to the `Cluster` objects which users can perform actual RPCs against.